internal/tfdiags/diagnostics.go

```diff
@@ -142,6 +142,28 @@ func diagnosticsForError(err error) []Diagnostic {
 	}
 }
 
+// Errors returns a Diagnostics list containing only diagnostics with a severity of Error.
+func (diags Diagnostics) Errors() Diagnostics {
```

Comment:
I am a bit hesitant to adding code which is not used anywhere other than in tests. Did you have any particular use case in mind? If not I would just remove it.

Reply:
Thanks for bringing that up @radeksimko! My intention was to add this for two reasons: 1) to provide consistency in the API, ensuring that equivalent methods for specific severities are available (as `tfdiags` seems to be imported across more projects, and 2) I spotted a few places (e.g.,) https://github.com/hashicorp/terraform/blob/ef168cb25ce5019f62aba7d92fac17d090301847/internal/terraform/graph.go#L62-L64 in the codebase where this new method could be useful. However, I’d prefer to make those changes in a separate PR. Does that make sense?